test/PositionValue.spec.ts

```diff
@@ -0,0 +1,224 @@
+      expect(principal.amount1).to.equal(0)
+    })
+
+    it('returns the correct values when range is skewed above price', async () => {
+      await nft.mint({
+        token0: tokens[0].address,
+        token1: tokens[1].address,
+        tickLower: -6_000,
+        tickUpper: getMaxTick(TICK_SPACINGS[FeeAmount.MEDIUM]),
+        fee: FeeAmount.MEDIUM,
+        recipient: wallets[0].address,
+        amount0Desired: amountDesired,
+        amount1Desired: amountDesired,
+        amount0Min: 0,
+        amount1Min: 0,
+        deadline: 10,
+      })
+
+      const principal = await positionValue.principal(nft.address, 1)
+      expect(principal.amount0).to.equal(14)
+      expect(principal.amount1).to.equal(3)
+    })
+
+    it('returns the correct values when range is skewed below price', async () => {
+      await nft.mint({
+        token0: tokens[0].address,
+        token1: tokens[1].address,
+        tickLower: getMinTick(TICK_SPACINGS[FeeAmount.MEDIUM]),
+        tickUpper: 6_000,
+        fee: FeeAmount.MEDIUM,
+        recipient: wallets[0].address,
+        amount0Desired: amountDesired,
+        amount1Desired: amountDesired,
+        amount0Min: 0,
+        amount1Min: 0,
+        deadline: 10,
+      })
+
+      const principal = await positionValue.principal(nft.address, 1)
+      expect(principal.amount0).to.equal(3)
+      expect(principal.amount1).to.equal(14)
+    })
+  })
+
+  describe('#fees', () => {
+    it.only('return the correct amount of fees', async () => {
+      await nft.mint({
+        token0: tokens[0].address,
+        token1: tokens[1].address,
+        tickLower: getMinTick(TICK_SPACINGS[FeeAmount.MEDIUM]),
+        tickUpper: getMaxTick(TICK_SPACINGS[FeeAmount.MEDIUM]),
+        fee: FeeAmount.MEDIUM,
+        recipient: wallets[0].address,
+        amount0Desired: amountDesired,
+        amount1Desired: amountDesired,
+        amount0Min: 0,
+        amount1Min: 0,
+        deadline: 10,
+      })
+
+      const swapAmount = 10000
+      await tokens[0].approve(router.address, swapAmount)
+      await router.exactInput({
+        recipient: wallets[0].address,
+        deadline: 1,
+        path: encodePath([tokens[0].address, tokens[1].address], [FeeAmount.MEDIUM]),
+        amountIn: swapAmount,
+        amountOutMinimum: 0,
+      })
+
+      // this works
+      const fees = await nft.callStatic.collect({
+        tokenId: 1,
+        recipient: wallets[0].address,
+        amount0Max: MaxUint128,
+        amount1Max: MaxUint128,
+      })
+      // console.log(fees.amount0.toString())
+      // console.log(fees.amount1.toString())
+
+      const pool = new ethers.Contract(
+        computePoolAddress(await nft.factory(), [tokens[0].address, tokens[1].address], FeeAmount.MEDIUM),
+        IUniswapV3PoolABI,
+        wallets[0]
+      )
+
+      // this reverts with "NP"
+      await pool.burn(getMinTick(TICK_SPACINGS[FeeAmount.MEDIUM]), getMaxTick(TICK_SPACINGS[FeeAmount.MEDIUM]), 0, {
```

Comment:
I need to update the pool position to get up-to-date fee amounts. Any idea why collect (L200) works just fine, but I fall into a revert when I burn 0 liquidity here?? (cc @moodysalem )

Reply:
position owner is the nft, you can't burn it from a wallet

Reply:
this is probably the trickiest part of computing fees, unfortunately

Reply:
just to elaborate, since i came back and forgot what i was talking about

pool.burn is being called from a wallet, which has no position at the min/max tick, so pool#burn throws NP for 'no position'
https://github.com/Uniswap/uniswap-v3-core/blob/f03155670ec1667406b83a539e23dcccf32a03bc/contracts/libraries/Position.sol#L54